webapp/src/components/user_selector/autocomplete_selector.jsx

```diff
@@ -4,151 +4,89 @@
-        } = this.props;
-
-        const {focused} = this.state;
-        let {input} = this.state;
-
-        if (!focused) {
-            input = value;
-        }
-
-        let labelContent;
-        if (label) {
-            labelContent = (
-                <label
-                    className={'control-label ' + labelClassName}
-                >
-                    {label}
-                </label>
-            );
-        }
-
-        let helpTextContent;
-        if (helpText) {
-            helpTextContent = (
-                <div className='help-text'>
-                    {helpText}
-                </div>
-            );
-        }
-
-        return (
-            <div
-                data-testid='autoCompleteSelector'
-                className='form-group'
-            >
-                {labelContent}
-                <div className={inputClassName}>
-                    <SuggestionBox
-                        placeholder={placeholder}
-                        ref={this.setSuggestionRef}
-                        listComponent={SuggestionList}
-                        className='form-control'
-                        containerClass='select-suggestion-container'
-                        value={input}
-                        onChange={this.onChange}
-                        onItemSelected={this.handleSelected}
-                        onFocus={this.onFocus}
-                        onBlur={this.onBlur}
-                        providers={providers}
-                        completeOnTab={true}
-                        renderNoResults={true}
-                        openOnFocus={true}
-                        openWhenEmpty={true}
-                        replaceAllInputOnSelect={true}
-                        disabled={disabled}
-                    />
-                    {helpTextContent}
-                    {footer}
-                </div>
+    let helpTextContent;
+    if (helpText) {
+        helpTextContent = (
+            <div className='help-text'>
+                {helpText}
             </div>
         );
     }
+
+    const getOptionData = (option) => option.username;
+
+    return (
+        <div
+            data-testid='autoCompleteSelector'
+            className='form-group'
+        >
+            {labelContent}
+            <div className={inputClassName}>
+                <AsyncSelect
+                    cacheOptions={true}
+                    loadOptions={loadOptions}
+                    defaultOptions={true}
+                    isClearable={true}
+                    disabled={disabled}
+                    placeholder={placeholder}
+                    getOptionLabel={getOptionData}
+                    getOptionValue={getOptionData}
+                    onChange={handleSelected}
+                />
+                {helpTextContent}
+                {footer}
+            </div>
+        </div>
+    );
 }
+
+AutocompleteSelector.propTypes = {
+    loadOptions: PropTypes.func.isRequired,
```

Comment:
Since this file is pretty small, do you mind writing this file in typescript? Here's some docs on this if you are not familiar with typescript https://developers.mattermost.com/contribute/webapp/migrating-to-typescript

Reply:
Sure, I'll get to that. 👍